src/core/sagas/product-details.ts

```diff
@@ -21,25 +21,29 @@ export namespace Tasks {
           refinements: [{ navigationName: 'id', type: 'Value', value: id }]
         }
       );
-
       if (records.length !== 0) {
         const [record] = records;
-        yield effects.call(receiveDetailsProduct, flux, record);
+        yield effects.put(flux.actions.setDetails(record));
       } else {
         throw new Error(`no records found matching id: ${id}`);
       }
     } catch (e) {
-      yield effects.put(flux.actions.receiveDetailsProduct(e));
+      yield effects.put(flux.actions.updateDetails(e));
     }
   }
 
-  export function* receiveDetailsProduct(flux: FluxCapacitor, record: Store.Product) {
-    flux.emit(Events.BEACON_VIEW_PRODUCT, record);
-    yield effects.put(flux.actions.receiveDetailsProduct(record.allMeta));
+  export function* receiveDetailsProduct(flux: FluxCapacitor, { payload: record }: Actions.SetDetails) {
+    let product = record;
+    if (record.allMeta) {
+      flux.emit(Events.BEACON_VIEW_PRODUCT, record);
+      product = record.allMeta;
+    }
+    yield effects.put(flux.actions.updateDetails(product));
```

Comment:
How about record = record.allmeta on line 39 and maybe change { payload: record } to { payload: product }